build_tools/cmake/easy.sh

```diff
@@ -0,0 +1,291 @@
+      arg_build=1
+      if [[ $i < $(($# - 1)) ]]
+      then
+        arg_targets="${args[@]:$((i+1))}"
+      fi  
+      break;;
+    *) echo "Error: unkown argument ${args[i]}" >&2; exit 1;;
+  esac
+done
+
+if [[ "${arg_printonly}" != 1 && -f "CMakeLists.txt" ]]
+then
+  echo "Error: We seem to be in a source directory. Please cd into a build directory." >&2
+  exit 1
+fi
+
+if [[ -z "${arg_build_type}" ]]
+then
+  # Sane default build type
+  # TODO: upstream this sane default to IREE's CMakeLists
+  arg_build_type=RelWithDebInfo
+fi
+
+function walk_up_to_iree_root() {
+  path="$(realpath -s "$1")"
+  while true
+  do
+    if [[ "$(basename "${path}")" == "iree" ]]
+    then
+      echo "${path}"
+      return
+    fi
+    if [[ -z "${path}" ]]
+    then
+      echo "Error: Could not find IREE root directory by walking up from $1" >&2
+      exit 1
+    fi
+    path="$(dirname "${path}")"
+  done
+}
+
+if [[ -z "${arg_src}" ]]
+then
+  # No explicit src directory specified. Infer from this script's location.
+  absolute_iree_dir="$(walk_up_to_iree_root "$0")"
+  relative_iree_dir="$(realpath -s --relative-to="$(pwd)" "${absolute_iree_dir}")"
+  if [[ -f "${relative_iree_dir}/CMakeLists.txt" ]]
+  then
+    arg_src="${relative_iree_dir}"
+  fi
+fi
+
+# Detect Linux builds (not including Android)
+if [[ "$OSTYPE" == "linux-gnu"* && -z "${arg_ndk}" ]]
+then
+  is_linux_build=1
+fi
+
+# Build a list of CMake variables to set.
+cmake_var_names=()
+cmake_var_values=()
+
+function add_cmake_var() {
+  if [[ ! -z "$1" ]]
+  then
+    cmake_var_names+=("$2")
+    cmake_var_values+=("$3")
+  fi
+}
+
+add_cmake_var "${arg_build_type}" CMAKE_BUILD_TYPE "${arg_build_type}"
+add_cmake_var "${arg_ndk}" CMAKE_TOOLCHAIN_FILE "${arg_ndk}/build/cmake/android.toolchain.cmake"
+add_cmake_var "${arg_ndk}" ANDROID_ABI "${android_abi}"
+add_cmake_var "${arg_ndk}" ANDROID_PLATFORM "${android_platform}"
+add_cmake_var "${arg_ndk}" IREE_HOST_C_COMPILER "$(which "$CC")"
+add_cmake_var "${arg_ndk}" IREE_HOST_CXX_COMPILER "$(which "$CXX")"
+add_cmake_var "${arg_ndk}" IREE_BUILD_COMPILER OFF
+add_cmake_var "${arg_ndk}" IREE_BUILD_SAMPLES OFF
+add_cmake_var "${arg_ccache}" IREE_ENABLE_CCACHE ON
+add_cmake_var "${arg_ccache}" LLVM_CCACHE_BUILD ON
+add_cmake_var "${arg_asan}" IREE_ENABLE_ASAN ON
+add_cmake_var "${arg_msan}" IREE_ENABLE_MSAN ON
+add_cmake_var "${arg_tsan}" IREE_ENABLE_TSAN ON
+add_cmake_var "${arg_tf}" IREE_BUILD_TENSORFLOW_COMPILER ON
+add_cmake_var "${arg_tflite}" IREE_BUILD_TFLITE_COMPILER ON
+add_cmake_var "${arg_xla}" IREE_BUILD_XLA_COMPILER ON
+add_cmake_var "${arg_py}" IREE_BUILD_PYTHON_BINDINGS ON
+add_cmake_var "${arg_java}" IREE_BUILD_JAVA_BINDINGS ON
+add_cmake_var "${arg_tracy}" IREE_ENABLE_RUNTIME_TRACING ON
+# On Linux, the default choice of compiler might be GCC, so we need to override
+# that.
+add_cmake_var "${is_linux_build}" CMAKE_C_COMPILER "$(which "$CC")"
+add_cmake_var "${is_linux_build}" CMAKE_CXX_COMPILER "$(which "$CXX")"
```

Comment:
A few people have built with gcc before, and we should support it if possible. We do expect most developers (who may use this script) to use clang though.

Reply:
that's why at the top of this file we have

```
# This syntax allows the user to override this by defining environment
# variables.
# Note that these are currently only used on Linux and Android builds, whence
# the lack of concern about hardcoding clang as default. Might still need to
# change this to support Android builds on Windows host?
: ${CC:=clang}
: ${CXX:=clang++}
```

However, we actually don't support GCC at all in some build configurations, at least in those that involve building TensorFlow.